### PR TITLE
Added new rule: component-no-proptypes

### LINKED ISSUE #3: New Rule: component-no-proptypes
Ban `propTypes` (TS should manage type checking, not React)

### PR DESCRIPTION
**Pull request by davidlipo**
Fixes #3

### CODE REVIEW

File: test/rules/component-no-proptypes/test.tsx.lint [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const test = "test";
+
+const propTypes = {};
+      ~~~~~~~~~ [PropTypes are forbidden as TypeScript should manage type checking, not React]
+
+propTypes: {
+~~~~~~~~~ [PropTypes are forbidden as TypeScript should manage type checking, not React]
+	test: React.PropTypes.string
```
Comment (adidahiya) [inline test/rules/component-no-proptypes/test.tsx.lint]: use spaces, not tabs, for indentation

File: test/rules/component-no-proptypes/test.tsx.lint [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const test = "test";
+
+const propTypes = {};
+      ~~~~~~~~~ [PropTypes are forbidden as TypeScript should manage type checking, not React]
```
Comment (adidahiya) [inline test/rules/component-no-proptypes/test.tsx.lint]: I don't think this should be an error.

**Comment by adidahiya** [comment]
If we're going to publish it, this rule needs to be smarter and only ban `propTypes` when they are actually used to do runtime type checking of component props. As it currently stands, it is too simplistic to be useful as a general rule. I wouldn't actually classify #3 as "Difficulty: Easy"

**Comment by JKillian** [comment]
I think a fair starting point for the rule would be to ban the use of `propTypes` as a static class member. It could also ban a `propTypes` key inside of a `React.createClass` function call. This would address the most common cases we want to prevent and be a reasonably simple implementation

**Comment by davidlipo** [comment]
Should I be continuing this with the simple implementation? Or is it not worth it?

**Comment by adidahiya** [comment]
@JKillian's suggestion above is good; you can continue with that approach but overall this rule is pretty low priority for us

**Comment by adidahiya** [comment]
closing as stale

**Pull request closed without merging**